### PR TITLE
CMake: add scope to target_link_libraries()

### PR DESCRIPTION
**Pull request by leecbaker**
This is necessary when using libzip as a submodule and using a newer version of CMake.

Fixes https://github.com/nih-at/libzip/issues/131

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -225,7 +225,7 @@ SET_TARGET_PROPERTIES(zip PROPERTIES VERSION 5.0 SOVERSION 5)
 ENDIF()
 
 
-TARGET_LINK_LIBRARIES(zip ${ZLIB_LIBRARIES} ${OPTIONAL_LIBRARY})
+TARGET_LINK_LIBRARIES(zip PUBLIC ${ZLIB_LIBRARIES} ${OPTIONAL_LIBRARY})
```
Comment (0-wiz-0) [inline lib/CMakeLists.txt]: Just for clarification: Are the (compression) libraries that libzip depends on really `PUBLIC`? None of their symbols are passed on to the outside via header files.

Comment (leecbaker) [reply lib/CMakeLists.txt]: As I understand it, PUBLIC mirrors the previous behaviour.

If symbols from the other libraries aren't visible from outside, then it sounds like `PRIVATE` might be the more correct way to go, but I'm not familiar enough with how this works to say for sure.

Comment (Flamefire) [reply lib/CMakeLists.txt]: > As I understand it, PUBLIC mirrors the previous behaviour.

Correct. But judging from the usage those libraries are only internally used, so they can be PRIVATE. Simple rule of thumb: If a dependent libraries headers are included in this libraries "public headers" (expected to be used by users), then use PUBLIC, else PRIVATE.

**Comment by leecbaker** [comment]
I've made a couple of changes; the zlib headers should no longer be included in the interface to the libzip library. It's ready for review again, if anybody wants to take a look.

**Comment by 0-wiz-0** [comment]
Addressed #131 differently, thanks!

**Pull request closed without merging**